server/shared/database/migrations/20200217104115-add-has_changes-to-repository.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+
+const head = require('lodash/head');
+const Promise = require('bluebird');
+const some = require('lodash/some');
+
+const TABLE_NAME = 'repository';
+const COLUMN_NAME = 'has_changes';
+
+const isModified = it => it.modifiedAt > it.publishedAt;
+
+exports.up = async (qi, { BOOLEAN }) => {
+  await qi.addColumn(TABLE_NAME, COLUMN_NAME, { type: BOOLEAN });
+  await updateHasChanges(qi);
+};
+
+exports.down = queryInterface => {
+  return queryInterface.removeColumn(TABLE_NAME, COLUMN_NAME);
+};
+
+async function updateHasChanges({ sequelize }) {
+  const t = await sequelize.transaction();
+  const repositories = await getRepositories(sequelize, t);
+  await Promise.map(repositories, async ({ id }) => {
+    const activities = await getOutlineActivities(sequelize, id, t);
```

Comment:
Might be confusing / one might expect empty array if there are none (commenting because following line)

Reply:
Any suggestions on how to improve this?

Reply:
Just make sure getOutlineActivities always returns an array